apps/judicial-system/web/src/routes/Prison/IndictmentOverview/IndictmentOverview.tsx

```diff
@@ -15,19 +16,46 @@ import {
   PageTitle,
   RenderFiles,
 } from '@island.is/judicial-system-web/src/components'
-import { CaseFileCategory } from '@island.is/judicial-system-web/src/graphql/schema'
-import { useFileList } from '@island.is/judicial-system-web/src/utils/hooks'
+import {
+  CaseFileCategory,
+  Defendant,
+} from '@island.is/judicial-system-web/src/graphql/schema'
+import {
+  useCase,
+  useDefendants,
+  useFileList,
+} from '@island.is/judicial-system-web/src/utils/hooks'
 
 import { strings } from './IndictmentOverview.strings'
+import { PunishmentType } from '@island.is/judicial-system/types'
 
 const IndictmentOverview = () => {
   const { workingCase } = useContext(FormContext)
   const { formatMessage } = useIntl()
+  const { limitedAccessUpdateDefendant } = useDefendants()
 
   const { onOpen } = useFileList({
     caseId: workingCase.id,
   })
 
+  const { defendants } = workingCase
+  const defendant =
+    defendants && defendants?.length > 0 ? defendants[0] : undefined
+  const [selectedPunishmentType, setPunishmentType] = useState<PunishmentType>()
+
+  const onChange = (updatedPunishmentType: PunishmentType) => {
+    defendant &&
+      limitedAccessUpdateDefendant({
+        caseId: workingCase.id,
+        defendantId: defendant.id,
+        punishmentType: updatedPunishmentType,
+      })
+    setPunishmentType(updatedPunishmentType)
+  }
+
+  const hasSetPunishmentType = (punishmentType: PunishmentType) =>
```

Comment:
When a `punishmentType` value is already stored in the db, by default the radio button selection is empty